app/services/registration_transfer_service.rb

```diff
@@ -27,18 +19,56 @@ def find_transient_registration
   def find_user(email)
     return nil unless email.present?
 
-    ExternalUser.where(email: email).first
+    return unless ExternalUser.where(email: email).first
+
+    # If a matching user exists, transfer and return a success status
+    update_account_emails(email)
+    send_existing_account_confirmation_email
+
+    :success_existing_user
+  end
+
+  def create_new_user(email)
+    return nil unless email.present?
+
+    token = invite_user_and_return_token(email)
+    return unless token.present?
+
+    # If a new user is created, transfer and return a success status
+    update_account_emails(email)
+    send_new_account_confirmation_email(token)
+
+    :success_new_user
+  end
+
+  def invite_user_and_return_token(email)
+    ExternalUser.invite!(email: email, skip_invitation: true).raw_invitation_token
+  end
+
+  def update_account_emails(email)
+    update_account_email_for(@registration, email)
+    update_account_email_for(@transient_registration, email) if @transient_registration.present?
   end
 
-  def update_account_email_for(registration)
-    registration.account_email = @recipient_user.email
+  def update_account_email_for(registration, email)
+    registration.account_email = email
     registration.save!
   end
 
-  def send_confirmation_email
+  def send_existing_account_confirmation_email
     RegistrationTransferMailer.transfer_to_existing_account_email(@registration).deliver_now
   rescue StandardError => e
-    Airbrake.notify(e) if defined?(Airbrake)
-    Rails.logger.error "Registration transfer confirmation email error: " + e.to_s
+    log_email_error(e)
+  end
+
+  def send_new_account_confirmation_email(token)
+    RegistrationTransferMailer.transfer_to_new_account_email(@registration, token).deliver_now
+  rescue StandardError => e
+    log_email_error(e)
+  end
+
+  def log_email_error(error)
+    Airbrake.notify(error) if defined?(Airbrake)
```

Comment:
I'd like to see us start sending additional data such as the registration number when we send an airbrake notification
```suggestion
    Airbrake.notify(error, reg_identifier: @registration.reg_identifier) if defined?(Airbrake)
```
See https://github.com/airbrake/airbrake-ruby#airbrakenotify